src/store/categories.js

```diff
@@ -22,27 +23,73 @@ export const showCategory = () => {
   }
 }
 
-export const displayProducts = (category) => {
+export const saveProducts = (category) => {
+  console.log('name of category', category)
   return {
-    type: 'DISPLAY',
-    payload: category,
+    type: 'SAVE',
+    payload: category.name,
   }
 }
 
-let categorySwitchboard = (state = initialState, action) => {
-  console.log(state);
-  let { type } = action;
+export const addToCart = (product) => {
+  console.log(product);
+  return {
+    type: 'ADDTOCART',
+    payload: product.name
+  }
+}
+
+export const removeFromCart = (product) => {
+  return {
+    type: 'REMOVEFROMCART',
+    payload: product.name
+  }
+}
+
+// eslint-disable-next-line import/no-anonymous-default-export
+export default (state = initialState, action) => {
+  let { type, payload } = action;
+  console.log(action)
+  if (!action.payload) {
+    console.log(state, action);
+  } else {
+    console.log(action)
+  }
   switch (type) {
     case 'CATEGORIES':
 
       return initialState;
 
+    case 'SAVE':
+
+      return {
+        ...state,
+        activeCategory: payload,
+
+      };
+
+    case 'ADDTOCART':
+
+      return {
+        ...state,
+        cart: payload
+      };
+
+    case 'REMOVEFROMCART':
+
+      return {
+        ...state,
+        cart: payload,
+
+      };
+
+
     case 'DISPLAY':
       return initialState;
 
 
     default:
+
       return state;
   }
 }
-export default categorySwitchboard;
```

Comment:
-1 point I would like to see more added to that README that isn't just autogenerated
-1 point when I click on a product I get an error when I hit the cart
-1 point I would like to see inventory information rendered onto the page